src/lang/parts/IndexFindersPart.php

```diff
@@ -15,16 +15,16 @@ trait IndexFindersPart {
 	abstract public function findLast(...$arguments);
 
 	/**
-	 * Returns the index of the given element or false if the element can't be found
+	 * Returns the index of the given element or null if the element can't be found
 	 *
 	 * @param mixed $element
 	 *
-	 * @return int the index for the given element
+	 * @return int|string|null the index for the given element
 	 */
-	public function indexOf($element): ?int {
+	public function indexOf($element) {
 		$out = array_search($element, $this->array, true);
 
-		return false === $out ? null : (int) $out;
+		return false === $out ? null : $out;
```

Comment:
Please no yoda style... hard to read it is.

Reply:
fixed